library/job.go

```diff
@@ -48,6 +50,21 @@ var _ gitcollector.Job = (*Job)(nil)
 // JobFn represents the task to be performed by a Job.
 type JobFn func(context.Context, *Job) error
 
+// TODO: we should probably secure other fiels
+func (j *Job) SetEndpoints(endpoints []string) {
+	j.mu.Lock()
+	defer j.mu.Unlock()
+
+	j.Endpoints = endpoints
+}
+
+func (j *Job) GetEndpoints() []string {
```

Comment:
If we we are going to have getter and setter I would unexport `Endpoints` and rename `GetEndpoints` to `Endpoints`.

Reply:
Done.